src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs

```diff
@@ -411,71 +412,381 @@ public async Task AcceptToMemoryPool_WithSegWitValidTxns_IsSuccessfullAsync()
-            // - Check input scriptsig push only
+            string dataDir = GetTestDirectoryPath(this);
+
+            // Run mempool tests on mainnet so that RequireStandard flag is set in the mempool settings.
+            Network network = KnownNetworks.Main;
+            var minerSecret = new BitcoinSecret(new Key(), network);
+            ITestChainContext context = await TestChainFactory.CreateAsync(network, minerSecret.PubKey.Hash.ScriptPubKey, dataDir);
+            IMempoolValidator validator = context.MempoolValidator;
+            Assert.NotNull(validator);
+
+            var tx = new Transaction();
+            tx.AddInput(new TxIn(new OutPoint(context.SrcTxs[0].GetHash(), 0), PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(minerSecret.PubKey)));
+
+            // Add outputs until transaction is large enough to fail standardness checks.
+            while (MempoolValidator.GetTransactionWeight(tx, network.Consensus.Options) < this.Network.Consensus.Options.MaxStandardTxWeight)
+            {
+                tx.AddOutput(new TxOut(Money.Coins(1), minerSecret.PubKeyHash));
+            }
+
+            // It is not necessary to sign the transaction as we are testing a standardness
+            // requirement which is evaluated before signatures in the mempool.
+
+            var state = new MempoolValidationState(false);
+
+            // Tests the transaction size standardness check in PreMempoolChecks CheckStandardTransaction
+            bool isSuccess = await validator.AcceptToMemoryPool(state, tx);
+            Assert.False(isSuccess, "Transaction that is too large should not have been accepted.");
+            Assert.Equal(MempoolErrors.TxSize, state.Error);
         }
 
-        [Fact(Skip = "Not implemented yet.")]
-        public void AcceptToMemoryPool_TxIsNonStandardScriptTemplateIsNull_ReturnsFalse()
+        [Fact]
+        public async void AcceptToMemoryPool_TxIsNonStandardInputScriptSigsLengthInvalid_ReturnsFalseAsync()
         {
-            // TODO:Test the casses in PreMempoolChecks CheckStandardTransaction
-            // - Check output scripttemplate == null
+            string dataDir = GetTestDirectoryPath(this);
+
+            // Run mempool tests on mainnet so that RequireStandard flag is set in the mempool settings.
+            Network network = KnownNetworks.Main;
+            var minerSecret = new BitcoinSecret(new Key(), network);
+            ITestChainContext context = await TestChainFactory.CreateAsync(network, minerSecret.PubKey.Hash.ScriptPubKey, dataDir);
+            IMempoolValidator validator = context.MempoolValidator;
+            Assert.NotNull(validator);
+
+            var destSecret = new BitcoinSecret(new Key(), network);
+            var tx = new Transaction();
+
+            Op checkSigOp = new Op() { Code = OpcodeType.OP_CHECKSIG };
+
+            List<Op> sigOps = new List<Op>();
+
+            // A scriptSig should not be longer than 10 000 bytes.
```

Comment:
This is a bit confusing you are adding the op itself `OP_CHECKSIG` 10k times, hmm I am not sure how the rule check is but you should create a sig data that is too big no?

Reply:
Well, if I remember correctly the size check precedes a lot of the others, so it doesn't actually make a difference what type of data is in the sig. That is a common difficulty in these tests, trying to test the exact code path that throws a particular error. I will see if tweaking the contents of the scriptSig will still give the desired result.